index.js

```diff
@@ -81,79 +106,135 @@ ${details("replace", resources_to_replace, "+")}
 ${commentFooter.map(a => a == '' ? '\n' : a).join('\n')}
 ${workflowLink}
 `
-                if (resources_to_create + resources_to_delete + resources_to_update + resources_to_replace == []) {
-                    hasNoChanges = true;
-                }
-            } else {
-                hasNoChanges = true;
-                console.log("No changes found in the plan. setting hasNoChanges to true.")
-                body += `
+        if (resources_to_create + resources_to_delete + resources_to_update + resources_to_replace == []) {
+          hasNoChanges = true;
+        }
+      } else {
+        hasNoChanges = true;
+        console.log("No changes found in the plan. setting hasNoChanges to true.")
+        body += `
 <p>There were no changes done to the infrastructure.</p>
 `
-                core.info(`"The content of ${file} did not result in a valid array or the array is empty... Skipping."`)
-            }
-        } catch (error) {
-            core.error(`${file} is not a valid JSON file. error: ${error}`);
-        }
+        core.info(`"The content of ${file} did not result in a valid array or the array is empty... Skipping."`)
+      }
+    } catch (error) {
+      core.error(`${file} is not a valid JSON file. error: ${error}`);
     }
-    return body;
+  }
+  return body;
 }
 
 const details = (action, resources, operator) => {
-    let str = "";
+  let str = "";
 
-    if (resources.length !== 0) {
-        str = `
+  if (resources.length !== 0) {
+    str = `
 #### Resources to ${action}\n
 \`\`\`diff\n
 `;
-        for (const el of resources) {
-            // In the replace block, we show delete (-) and then create (+)
-            if (action === "replace") {
-                str += `- ${el}\n`
-            }
-            str += `${operator} ${el}\n`
-        }
-
-        str += "```\n"
+    for (const el of resources) {
+      // In the replace block, we show delete (-) and then create (+)
+      if (action === "replace") {
+        str += `- ${el}\n`
+      }
+      str += `${operator} ${el}\n`
     }
 
-    return str;
+    str += "```\n"
+  }
+
+  return str;
 }
 
+const queryComments = (variables) => {
+    return octokit.graphql(commentsQuery, variables);
+};
+
+const minimizeComment = (variables) => {
+    return octokit.graphql(minimizeCommentQuery, variables);
+};
+
+const hideComments = () => {
+    core.info(`Hiding previous comments.`);
+
+    queryComments({
+      owner: context.repo.owner,
+      name: context.repo.repo,
+      number: context.issue.number
+    })
+      .then(response => {
+        core.info(`Successfully retrieved comments for PR #${context.issue.number}.`);
+        const comments = response.repository.pullRequest.comments.nodes;
+
+        core.info(`Found ${comments.length} comments in the PR.`);
+
+        const filteredComments = comments.filter(comment =>
+          comment.body.includes('Terraform Plan:') ||
```

Comment:
@DeWildeDaan Should this check for the `commentHeader` in case the default value is not used?